arrangor/src/main/kotlin/no/nav/amt/tiltak/arrangor/ArrangorServiceImpl.kt

```diff
@@ -46,8 +49,30 @@ class ArrangorServiceImpl(
 		return arrangorRepository.getByOrganisasjonsnummer(virksomhetsnummer)?.toArrangor()
 	}
 
-	override fun getVirksomheterForAnsatt(ansattId: UUID): List<Arrangor> {
-		throw NotImplementedError("getVirksomheterForAnsatt in ArrangorService is not yet implemented")
+	override fun oppdaterArrangor(arrangorUpdate: ArrangorUpdate) {
+		val arrangor = getArrangorByVirksomhetsnummer(arrangorUpdate.organisasjonsnummer)
+
+		if (arrangor == null) {
+			// Sjekk om virksomheten er en overordnet enhet til noen arrangører og oppdater navn.
+			arrangorRepository.updateOverordnetEnhetNavn(arrangorUpdate.organisasjonsnummer, arrangorUpdate.navn)
+			return
+		}
+
+		val overordnetEnhet = arrangorUpdate.overordnetEnhetOrganisasjonsnummer?.let {
+			enhetsregisterClient.hentVirksomhet(it)
+		}
+
+		transactionTemplate.executeWithoutResult {
+			arrangorRepository.update(
+				ArrangorUpdateDbo(
+					id = arrangor.id,
+					navn = arrangorUpdate.navn,
+					overordnetEnhetNavn = overordnetEnhet?.navn,
+					overordnetEnhetOrganisasjonsnummer = overordnetEnhet?.organisasjonsnummer,
+				)
+			)
+			arrangorRepository.updateOverordnetEnhetNavn(arrangorUpdate.organisasjonsnummer, arrangorUpdate.navn)
```

Comment:
Hvis man uansett skal oppdatere overOrdnetEnhetNavn, uansett om `arrangor` er null eller ikke, så er det lettere å forstå flyten om det bare gjøres med en gang, en gang.

Reply:
Jeg er enig, grunnen til å gjøre det på to plasser er fordi jeg ønsker en transaksjon når man gjør flere operasjoner samtidig. Men det er kanskje ikke så viktig her..?